test/jdk/javax/swing/BasicMenuItemUI/bug4239714.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 1999, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4239714
+ * @summary Tests that BasicMenuItemUI.installComponent() is protected
```

Comment:
I doubt this test is useful… Wouldn't it be verified with JCK?

I had a similar doubt about another case where test used a new method that was added in 1.3.0 or an earlier release.

Reply:
Since this isn't a test for applet I think we should keep it for now

Reply:
> Since this isn't a test for applet I think we should keep it for now

What do you mean?

All the test does is it verifies the `installComponents` method is accessible in a subclass of `BasicMenuItemUI`. The [`installComponents`](https://docs.oracle.com/en/java/javase/21/docs/api/java.desktop/javax/swing/plaf/basic/BasicMenuItemUI.html#installComponents(javax.swing.JMenuItem)) method is protected since JDK 1.3, it's now a public API. Removing it requires a CSR.

Can it be accidentally removed? Can it be accidentally made package-private or private? *Very unlikely.* Therefore, we can save time by not running this test.

Reply:
the comment about applet is a bit out of context. Its from a comment I made off-line that if you were taking
a test that was written as an Applet and it explicitly a test for applets, that would be a good example of
a test to drop since JDK 22 doesn't support Applets, even thought the API is still there.

In this case, I agree I would not have bothered porting this test that basically corresponds to the signature tests,
but its also harmless, notwithstanding the very tiny cost of running this as a (now) headless test.
But since it was already done, I think we can keep it. It is a thousand times better than it was before in its old form.

Reply:
Thank you for clarification.

Let's keep it then.